modules/core/02-client/migrations/v7/genesis.go

```diff
@@ -0,0 +1,82 @@
+package v7
+
+import (
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+
+	"github.com/cosmos/ibc-go/v6/modules/core/02-client/types"
+	"github.com/cosmos/ibc-go/v6/modules/core/exported"
+)
+
+// MigrateGenesis accepts an exported IBC client genesis file and migrates it to:
+//
+// - Update solo machine client state protobuf definition (v2 to v3)
+// - Remove all solo machine consensus states
+// - Remove localhost client
+func MigrateGenesis(clientGenState *types.GenesisState, cdc codec.ProtoCodecMarshaler) (*types.GenesisState, error) {
```

Comment:
A lot of this work is ported from v1 legacy migrations. I cleaned up a lot of it, but there might be some more cleanup. Generally working with genesis isn't very clean though